dts/arm/cypress/psoc6-pinctrl.dtsi

```diff
@@ -10,10 +10,54 @@
 	soc {
 		pinctrl@40310000 {
 			/* instance, signal, port, pin, hsiom [, flag1, ... ] */
+			DT_CYPRESS_HSIOM(uart0,  rx,  0, 2, act_6, input-enable);
+			DT_CYPRESS_HSIOM(uart0,  tx,  0, 3, act_6, drive-push-pull);
```

Comment:
The `input-enable`, `drive-push-pull` properties are only required if the pin controller driver itself needs to configure it. This is rare. On most architectures it's the responsibility of the peripheral module, in this case UART, to configure pin as input or output. I.e. if the pin is connected to RX signal it will be automatically configured as input and there is no programmatic way to change it.

From Cypres PSoC-6 Technical Reference Manual:
> When the SCB I 2 C, UART, or SPI modes are
used, the SCB controls the interface pins digital output state
and output enable.

As expected, the pins should be automatically configured as input or output. The `input-enable`, `drive-push-pull` properties can be removed from DT.

Reply:
@mnkp ,

I think what you are referring is the SCB `on-chip I/O signals` on below diagram:

![image](https://user-images.githubusercontent.com/7027887/115596967-1af25e00-a2af-11eb-85e6-d0d8d87016a2.png)

The `I/O pads` are configured by `act_6` and flags define `drive mode`.
The above diagram shows connection from SBC UART to I/O pads. HSIOM (High Speed I/O Matrix) is responsible to route I/O pins to SCB block. Each pin should have their own `drive mode` proper configured.

The current driver configure that too at:
https://github.com/zephyrproject-rtos/zephyr/blob/26336061307c774d8c58b9487996dc78483bce4c/drivers/serial/uart_psoc6.c#L100-L106

In this case, CY_GPIO_DM_HIGHZ means `input-enable` and CY_GPIO_DM_STRONG_IN_OFF is equal to `drive-push-pull`.

Reply:
Yes, it looks like the relevant part. In the diagram above the signals `uart_rx_out_en` and `uart_tx_out_en` are hardwired to 0 and 1 respectively. There does not seem to be a programmatic way to change it. I believe the calls to 
```
 /* Configure pins for UART operation */ 
 Cy_GPIO_SetDrivemode(config->port, config->rx_num, CY_GPIO_DM_HIGHZ); 
 Cy_GPIO_SetDrivemode(config->port, config->tx_num, 
 	CY_GPIO_DM_STRONG_IN_OFF);
```
are not required, will have no effect.

That said, and I probably should have mentioned it before, I haven't been working with PSoC-6 SoCs before and don't have hardware to test it. My statements are based purely on reading the datasheet.

It would be best to check it on hardware. If the driver works correctly with `Cy_GPIO_SetDrivemode` lines removed then I'm right, if it doesn't then I'm wrong.

Reply:
Hi @mnkp ,

I removed `Cy_GPIO_SetDrivemode` from source code and serial port not works.

Reply:
Thanks, so the `input-enable`, `drive-push-pull` properties need to stay in DT.